yellowbox/extras/logstash.py

```diff
@@ -1,30 +1,263 @@
+        # Logstash service is not docker related. It cannot actually connect to
+        # the network. However, other containers connected to the network can
+        # connect to the service with docker's usual host
+        return [_docker_host_name]
+
+    def disconnect(self, network: Any):
+        """Does nothing. Conforms to YellowService interface."""
+        pass
+
+    def _filter_records(self, level: int) -> Iterable[Dict[str, Any]]:
+        """Filter records in the given level or above."""
+        return (record for record in self.records if
+                logging.getLevelName(record["level"]) >= level)
+
+    def assert_logs(self, level: Union[str, int]):
+        """Asserts that log messages were received in the given level or above.
+
+        Resembles unittest.assertLogs.
+
+        Args:
+            level: Log level by name or number
+
+        Raises:
+            AssertionError: No logs above the given level were received.
+        """
+        if not isinstance(level, int):
+            level = logging.getLevelName(level.upper())
 
-__all__ = ['LogstashService', 'LOGSTASH_DEFAULT_PORT']
+        # In rare cases it might be false, but shouldn't happen generally.
+        assert isinstance(level, int)
 
-LOGSTASH_DEFAULT_PORT = 5959
+        if not any(self._filter_records(level)):
+            raise AssertionError(f"No logs of level {logging.getLevelName(level)} "
+                                 f"or above were received.")
 
+    def assert_no_logs(self, level: Union[str, int]):
+        """Asserts that no log messages were received in the given level or above.
 
-class LogstashService(SingleContainerService, RunMixin):
-    def __init__(self, docker_client: DockerClient, image='logstash:7.8.1', **kwargs):
-        super().__init__(create_and_pull(
-            docker_client, image, publish_all_ports=True, detach=True,
-            ports={LOGSTASH_DEFAULT_PORT: None}
-        ), **kwargs)
+         Args:
+             level: Log level by name or number
 
-    def start(self, retry_spec: Optional[RetrySpec] = None):
-        # for now we just start the service, re-implementing this service is an ongoing ticket
-        return super().start(retry_spec)
+         Raises:
+             AssertionError: A log above the given level was received.
+         """
+        if not isinstance(level, int):
+            level = logging.getLevelName(level.upper())
 
-    def client_port(self):
-        return get_ports(self.container)[LOGSTASH_DEFAULT_PORT]
+        # In rare cases it might be false, but shouldn't happen generally.
+        assert isinstance(level, int)
 
-    def stop(self, signal='SIGKILL'):
-        # change in default
-        return super().stop(signal)
+        record = next(self._filter_records(level), None)
+        if record:
+            raise AssertionError(f"A log level {record['level']} was received. "
+                                 f"Message: {record['message']}")
```

Comment:
needs `local_url` and `container_url` methods/properties

Reply:
I'll add them as attributes